### PR TITLE
added create and update product feature

### PR DESCRIPTION
**Pull request by XinChenCSC**
created Role enum and added Role to user class
added Admin authentication to upsert (create and update) feature (only admin user is allowed to create and update product)
- created AuthorizedAdmin annotation (any method with AuthorizedAdmin annotation can only access by admin)
- created InvalidRoleException class to handle non-admin accounts to access upsert()

### CODE REVIEW

File: src/main/java/com/revature/controllers/ProductController.java [hunk 1]
```diff
@@ -40,11 +41,24 @@ public ResponseEntity<Product> getProductById(@PathVariable("id") int id) {
     }
 
     @Authorized
-    @PutMapping
+    @AuthorizedAdmin
+    @PutMapping("/create-update")
     public ResponseEntity<Product> upsert(@RequestBody Product product) {
+    	Optional<Product> curr_pd = productService.findById(product.getId());
```
Comment (sophiagavrila) [inline src/main/java/com/revature/controllers/ProductController.java]: @XinChenCSC Please rename `curr_pd` to `currPd` to comply with camel casing conventions for naming local variables.

Here is the link to the sonar cloud report of this code smell: 
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=4&id=Revature-Sophia-ECommerce-1376_e-commerce-backend&open=AYIj3ZtF1IANdIAAKhTC

**Review by sophiagavrila — CHANGES_REQUESTED** [review]
Must resolve code smell -  please see comment below:

File: src/main/java/com/revature/controllers/ProductController.java (hunk 1 above)
Comment (XinChenCSC) [reply src/main/java/com/revature/controllers/ProductController.java]: fixed it

**Review by Jheflin1995 — APPROVED** [review]

**Review by ktumlinson — APPROVED** [review]
LGTM

**Pull request merged**